hpcflow/archive/cloud/dropbox_cp.py

```diff
@@ -0,0 +1,369 @@
+                print(f"Could not upload file '{file.path.name}' at '{file.path.root}',"
+                      f" due to a Dropbox error. File Not archived.")
+                continue
+
+    def _archive_file(self, file: FileToUpload) -> Union[dropbox.files.Metadata, None]:
+        """Upload a file to a dropbox directory such that if the local file is newer than the
+        copy on Dropbox, the newer file overwrites the older file, and if the local file is
+        older than the copy on Dropbox, the local file is uploaded with an "auto-incremented"
+        name.
+
+        Note that if the modified times are different, the file still won't be uploaded
+        by dropbox if the file contents are identical. The point of checking for the
+        modified times is to save some time by not having Dropbox check file contents.
+
+        Parameters
+        ----------
+        file
+            `FileToUpload` object describing file to be uploaded
+
+        Returns
+        --------
+        FileMetadata of file if file uploaded. If file not uploaded return None.
+        """
+        # Get the last modified time of the file on Dropbox
+        dropbox_modified = self._get_dropbox_file_modified_time(file.dropbox_path)
+        if not dropbox_modified:
+            # File does not exist on dropbox
+            return self._choose_session_type(file)
+        if file.last_modified == dropbox_modified:
+            # If file on Dropbox has the same modified time as the client file then don't upload.
+            return None
+        elif file.last_modified > dropbox_modified:
+            # If the client file is newer than the one on dropbox then overwrite the one on dropbox
+            file.overwrite_mode = dropbox.dropbox.files.WriteMode('overwrite')
+
+        return self._choose_session_type(file)
+
+    def _choose_session_type(self, file: FileToUpload) -> dropbox.files.FileMetadata:
+        """Choose whether to upload the file as a simple upload or a session upload
+        depending on its size.
+
+        Parameters
+        ----------
+        file
+            `FileToUpload` object describing file to be uploaded
+        """
+
+        try:
+            if file.size <= CHUNK_SIZE:
+                return self.simple_upload(file)
+            else:
+                return self.session_upload(file)
+        except Exception as err:
+            raise CloudProviderError(f'Cloud provider error: {err}')
+
+    def session_upload(self, file: FileToUpload) -> dropbox.files.FileMetadata:
+        """Upload a file to Dropbox using an Upload session. This is required for files > 150 MB.
+        Though is probably beneficial for smaller files as it allows progress to be monitored as
+        the upload progresses.
+
+        Parameters
+        ----------
+        file
+            `FileToUpload` object describing file to be uploaded
+        """
+        progress = tqdm.tqdm(total=file.size, desc=f"Uploading file: {file.path.name}",
```

Comment:
This is a nice idea, but this method will be called non-interactively 99% of the time; so I am not sure how this would work when outputting to `stdout`?